lua/entities/gmod_tardis/modules/sh_falling.lua

```diff
@@ -0,0 +1,148 @@
+            end
+
+            if math.abs(ang.p) <= 45 and math.abs(ang.r) <= 45 and vel.z > 0 then
+                local newvel_x = math.Clamp(vel.x * 0.05,-30,30)
+                local newvel_y = math.Clamp(vel.y * 0.05,-30,30)
+
+                ph:SetVelocityInstantaneous(-up * 50 * phm)
+                ph:AddVelocity(Vector(newvel_x,newvel_y,0))
+
+                local newavel_x = math.Clamp(angv.x * 0.1,-300,300)
+                local newavel_y = math.Clamp(angv.y * 0.1,-300,300)
+                local newavel_z = math.Clamp(angv.z * 0.1,-300,300)
+                ph:SetAngleVelocityInstantaneous(Vector(newavel_x,newavel_y,newavel_z))
+            end
+        end
+    end)
+else
+    ENT:OnMessage("fall_landing", function(self, data, ply)
+        if not TARDIS:GetSetting("sound") then return end
+
+        local hard = data[1]
+        local snds = self.metadata.Exterior.Sounds
+
+        if TARDIS:GetSetting("flight-externalsound") then
+            if not hard then
+                self:EmitSound(snds.FlightLand)
+            else
+                self:EmitSound(snds.FlightFall)
+            end
+        end
+
+        if IsValid(self.interior) and TARDIS:GetSetting("flight-internalsound") then
+            local snds_i = self.metadata.Interior.Sounds
+            self.interior:EmitSound(snds_i.FlightLand or snds.FlightLand)
+        end
+    end)
+
+    ENT:AddHook("Think", "falling", function(self)
+        if self:GetData("free_movement")
+            and TARDIS:GetSetting("flight-externalsound")
+            and TARDIS:GetSetting("sound")
+        then
+            local p=math.Clamp(self:GetVelocity():Length()/125,0,15)
+            if self.fallsound and self.fallsound:IsPlaying() then
+                local ply=LocalPlayer()
+                local e=ply:GetViewEntity()
+                if not IsValid(e) then e=ply end
+                if e:EntIndex()==-1 then -- clientside prop
+                    local ext=ply:GetTardisData("exterior")
+                    if ext then
+                        e=ext
+                    else
+                        e=ply
+                    end
+                end
+                if ply:GetTardisData("exterior")==self and e==self.thpprop and ply:GetTardisData("outside") then
+                    self.fallsound:ChangePitch(95+p,0.1)
+                else
+                    local pos = e:GetPos()
+                    local spos = self:GetPos()
+                    local doppler = (pos:Distance(spos+e:GetVelocity())-pos:Distance(spos+self:GetVelocity()))/200
+                    self.fallsound:ChangePitch(math.Clamp(95+p+doppler,80,120),0.1)
+                end
+
+                self.fallsound:ChangeVolume(0.75 * p / 20)
+            else
+                self.fallsound = CreateSound(self, self.metadata.Exterior.Sounds.FlightFallWind)
+                self.fallsound:SetSoundLevel(90)
+                self.fallsound:ChangeVolume(0.75 * p / 20)
+                self.fallsound:Play()
```

Comment:
Letting go of a TARDIS after dragging it around at speed will cause it to make a pop sound for some reason